qprofiler/src/org/qcmg/qprofiler/bam/BamSummarizerMT.java

```diff
@@ -43,6 +42,27 @@
 import org.qcmg.qprofiler.report.SummaryReport;
 import org.qcmg.qprofiler.summarise.Summarizer;
 
+/**
+ * This class will be called only when option "--threads_consumer" is specified and value > 0;
+ * otherwise a single thread mode will be called, which is BamSummarizer.
+ * 
+ * Here A SingleProducer will be created if without option "--threads_producer" or value <= 1; 
```

Comment:
--threads-producer
How about: "Here A SingleProducer will be created if option "--threads_producer" has a value <= 1;"

Reply:
if "--threads-consumer" is null or 0, then the option "--threads-producer" will be ignored regardless it is specified or not.  Otherwise "--threads-producer" will be applied, the default value is 1 if it is not specified or <=1.

Reply:
see details on code  inside Qprofiler.java
```
// get no of threads
noOfConsumerThreads = options.getNoOfConsumerThreads();
noOfProducerThreads = Math.max(1, options.getNoOfProducerThreads());
```

Reply:
Maybe that should make it into the usage so that the user is aware of this.